### PR TITLE
Modify assert!(match {}) calls in test cases to use assert_matches!()…

### PR DESCRIPTION
**Pull request by theyoyojo**
… from crate::assert_matches.

One notable exception: src/thindev.rs:test_basic().
Explanation of the workaround can be found inline at src/thindev.rs:543

### CODE REVIEW

**Comment by mulkieran** [comment]
@theyoyojo run "make fmt" on your src and resubmit. Make sure you're using rustc 1.31.0, it's the preferred development compiler.

**Comment by theyoyojo** [comment]
Unable to find rust 1.31.0 rpm for Fedora 29 x86_64. Using 1.32.0-1.fc29 instead.

**Comment by trgill** [comment]
add to whitelist

**Comment by theyoyojo** [comment]
@trgill which whitelist?

**Comment by mulkieran** [comment]
> @trgill which whitelist?

If you're on the Jenkins whitelist, then Jenkins tests will be run against your PRs.

**Comment by theyoyojo** [comment]
> > @trgill which whitelist?
> 
> If you're on the Jenkins whitelist, then Jenkins tests will be run against your PRs.

Ah ok

File: src/thindev.rs [hunk 1]
```diff
@@ -542,21 +542,24 @@ mod tests {
         assert_eq!(table.params.pool, tp.device());
         assert_eq!(table.params.thin_id, thin_id);
 
-        assert!(match td.status(&dm).unwrap() {
+        // Note: assert_matches cannot do a negative assertion, thus the following workaround
```
Comment (mulkieran) [inline src/thindev.rs]: Please drop this change. The previous version is more direct and gives the same amount of information.

**Comment by trgill** [comment]
@theyoyojo welcome to Stratis. 

FYI - Jenkins has a plugin that monitors comments to pull requests.  That comment was a command to the plugin.  

Please let me know if I can be any help.

**Comment by mulkieran** [comment]
I put the PR on the Stratis board: https://github.com/orgs/stratis-storage/projects/2.

**Comment by theyoyojo** [comment]
Thanks to @tasleson on the #rust channel, I was eventually able to properly install rust-1.31.0 via: https://koji.fedoraproject.org/koji/buildinfo?buildID=1170109

File: src/thindev.rs (hunk 1 above)
Comment (theyoyojo) [reply src/thindev.rs]: Done.

**Comment by theyoyojo** [comment]
@mulkieran  requested changes have been made.

**Comment by theyoyojo** [comment]
Done.

**Comment by theyoyojo** [comment]
Reopened as #439 Friday, just leaving a note on this PR for doc purposes

**Pull request closed without merging**